internal/metrics/collector/settings.go

```diff
@@ -62,15 +68,47 @@ func createConverterFactories() []confmap.ConverterFactory {
 }
 
 func createURIs(cfg *config.Config) []string {
-	return []string{getConfig(cfg)}
+	return []string{cfg.Metrics.CollectorConfigPath}
 }
 
-func getConfig(_ *config.Config) string {
-	val, ex := os.LookupEnv("OPENTELEMETRY_COLLECTOR_CONFIG_FILE")
-	if !ex {
-		return "/tmp/otel-collector-config.yaml"
+// Generates a OTel Collector config to a file by injecting the Metrics Config to a Go template.
+func writeCollectorConfig(conf *config.Metrics) error {
+	otelcolTemplate, err := template.New(otelTemplatePath).ParseFiles(otelTemplatePath)
+	if err != nil {
+		return err
+	}
+
+	// Should check that we are within an allowed directory in the future.
```

Comment:
Can you make sure to add a ticket to capture this future work in case it is forgotten about

Reply:
This is now checked when validating the `Collector` config. Moving forward we should validate all paths in our `Config` in a similar fashion.

I will add a ticket for this.

Reply:
Ticket added ✅